bloom/generators/common.py

```diff
@@ -139,9 +140,19 @@ def package_conditional_context(ros_distro):
     else:
         error("Bloom cannot cope with distribution_type '{0}'".format(
             distribution_type), exit=True)
+    python_version = get_python_version(ros_distro)
+    if python_version == 2:
+        ros_python_version = '2'
+    elif python_version == 3:
+        ros_python_version = '3'
+    else:
+        error("Bloom cannot cope with python_version '{0}'".format(
+            python_version), exit=True)
```

Comment:
Since the rosdistro index may not contain the optional `python_version` value, this could result in "Bloom cannot cope with python_version 'None'`, right? Is that intended behavior?

Reply:
I think that bloom needs to require the field because the silent failure first reported in https://github.com/ros-visualization/python_qt_binding/issues/76 is really difficult to trace for a downstream that isn't aware of the conditional dependency system and a package maintaner who isn't familiar with bloom's internals.

I think we could add a separate error message for the None case but I don't think we can set any fallback value that would allow the release to continue.

I think an interface from catkin_pkg that allows the evaluation to raise an error when a condition is undefined would allow us to only raise an error if a conditional is evaluated and not present, but that's not something that can be easily introduced.

You're right to point out that many other rosdistro indices probably haven't added this field yet. So a Discourse announcement concurrent with the release of this change is probably merited.

Reply:
> I think we could add a separate error message for the None case

I think that would be desirable. Just something to clarify that there isn't anything the maintainer should do differently, and that the problem is in the distribution file.